keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/MemoryCloudDriverCache.kt

```diff
@@ -57,7 +57,6 @@ class MemoryCloudDriverCache(
           }
       }
         .handleNotFound()
-        ?: notFound("Security group with id $id not found in the $account account and $region region")
```

Comment:
What's the cache behavior when `handleNotFound()` returns null? Does it keep the cache unpopulated, or does it cache a miss (`null`)?

Reply:
Keeps the cache unpopulated